contracts/buyAndBurn/BuyAndBurnSwapImplementation.sol

```diff
@@ -0,0 +1,402 @@
+            // NO PATH: Some TokenIn unused - quote was partial
+            // Use perSwapSlippageCap for tighter slippage control on partial swap
+
+            // Approve swapHelper if tokenIn is ERC20
+            if (params.tokenIn != VANA) {
+                IERC20(params.tokenIn).forceApprove(address(swapHelper), amountSwapIn);
+            }
+
+            // Execute swap with perSwapSlippageCap (tighter protection for partial swap)
+            (uint256 amountSwapInUsed, uint256 amountReceived) = swapHelper.slippageExactInputSingle{
+                    value: params.tokenIn == VANA ? amountSwapIn : 0
+                }(
+                ISwapHelper.SlippageSwapParams({
+                    tokenIn: params.tokenIn,    // Use original address (VANA or ERC20)
+                    tokenOut: params.tokenOut,  // Use original address (VANA or ERC20)
+                    fee: params.fee,
+                    recipient: address(this),
+                    amountIn: amountSwapIn,
+                    maximumSlippagePercentage: params.perSwapSlippageCap
+                })
+            );
+            amountSwapIn = amountSwapInUsed;
+            amountSwapOut = amountReceived;
+        }
+
+        // If we reach here, we have leftover tokenIn to handle
+
+        // Calculate leftover tokenIn (what wasn't swapped)
+        uint256 amountLpIn = params.amountIn - amountSwapIn;
+
+        // Wrap VANA to WVANA for LP if tokenIn is native VANA
+        if (params.tokenIn == VANA && amountLpIn > 0) {
+            WVANA.deposit{value: amountLpIn}();
+        }
+
+        // Handle WVANA wrapping for tokenOut based on what SwapHelper returned
+        if (params.tokenOut == VANA && amountSwapOut > 0) {
+            // Check if swap returned WVANA or native VANA
+            uint256 wvanaBalanceAfter = IERC20(address(WVANA)).balanceOf(address(this));
+            uint256 wvanaReceived = wvanaBalanceAfter > wvanaBalanceBefore ? wvanaBalanceAfter - wvanaBalanceBefore : 0;
+
+            // If we received less WVANA than expected, wrap the native VANA we received
+            if (wvanaReceived < amountSwapOut) {
+                uint256 ethToWrap = amountSwapOut - wvanaReceived;
+                require(address(this).balance >= ethToWrap, "Insufficient ETH received from swap");
+                WVANA.deposit{value: ethToWrap}();
+            }
+        }
+
+        // Try to add liquidity with leftover tokenIn (if any)
+        if (amountLpIn > 0) {
+            // Determine token0 and token1 based on swap direction
+            address token0 = zeroForOne ? tokenIn : tokenOut;
+            address token1 = zeroForOne ? tokenOut : tokenIn;
+
+            // We only add the leftover tokenIn to LP (greedy strategy prioritizes burn over LP)
+            // Uniswap V3 can handle one-sided liquidity when price is outside range
+            uint256 amount0Desired = zeroForOne ? amountLpIn : 0;
+            uint256 amount1Desired = zeroForOne ? 0 : amountLpIn;
```

Comment:
In this two lines, either `amount0Desired` or `amount1Desired` is 0 and there is no LP added due to that.